fah_alchemy/storage/statestore.py

```diff
@@ -830,11 +851,17 @@ def dequeue_taskqueue_tasks(
             // delete connections between node to remove and behind, ahead nodes
             DELETE behindf, aheadf
 
+            RETURN task
```

Comment:
Am I correct in thinking we only delete the relationship but leave the node intact incase there is another task depending on it? Just checking my understanding.

Reply:
In general we won't ever delete tasks, exactly for that reason, and also because we attach our `ProtocolDAGResultRef`s to them on completion or failure.